### DISCUSSION TITLE
How is Best Possible Recall computed?

### DISCUSSION THREAD

**Discussion by amdenisaionascu** [ultralytics/yolov5#8187] [Q&A]
I cam across the term Best Possible Recall while working with AutoAnchor and I have tried to understand how it works from the code. 
Does anyone have any resources which explain the implementation and the reasoning behind it? (I have tried to google it, but did not find much.)

My understanding is the following:
1. The gt are resized to account for the letterbox transformation
2. You compute the division between each combination of gt and anchors
3. Choose the anchor which gives the best score 
4. BPR = the proportion of gts which have a match better than a threshold (eg: 98.6% of gt match an anchor)

Thank you!

**Comment by glenn-jocher** [comment]
@amdenisaionascu see L40:
https://github.com/ultralytics/yolov5/blob/6a67594cbfe6599aab7006e33948f073e14246a8/utils/autoanchor.py#L35-L42
